src/libical/icalrecur.c

```diff
@@ -1959,13 +1959,29 @@ static int __day_diff(icalrecur_iterator *impl, icaltimetype a, icaltimetype b)
         diff = get_day_of_year(impl, b.year, b.month, b.day, NULL) -
             get_day_of_year(impl, a.year, a.month, a.day, NULL);
     } else {
+        /* Swap a and b if a is greater than b */
+        int flipped = 0;
```

Comment:
in order to make MSVC happy, you probably have to move the declaration for 'int flipped' near the top of the function.